tests/modules/code_provider/code_provider_service_test.py

```
@@ -0,0 +1,326 @@
+            assert "Error processing file content" in excinfo.value.detail
+
+    @pytest.mark.parametrize("token_list", ["", " ", ",", " , "])
+    @patch('os.getenv')
+    def test_init_github_no_tokens_raises_value_error(self, mock_getenv, mock_LocalRepoService, mock_GithubService, mock_sql_db, token_list):
+        """
+        Raise ValueError if 'GH_TOKEN_LIST' is empty, whitespace, or just commas in Production
+        """
+        mock_GithubService.side_effect = GithubService 
+        mock_getenv.return_value = token_list
+        with patch.dict(os.environ, {}, clear=True):
+             with pytest.raises(ValueError, match="GitHub token list is empty or not set in environment variables"):
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)